polygeist/tools/cgeist/Test/Verification/pair.c

```diff
@@ -18,25 +18,30 @@ int create() {
   return p2.a;
 }
 
-// CHECK:   func @byval(%arg0: memref<?x2xi32>, %arg1: i32, %arg2: memref<?x2xi32>)
-// CHECK-NEXT:     affine.store %arg1, %arg0[0, 1] : memref<?x2xi32>
-// CHECK-NEXT:     %0 = affine.load %arg0[0, 0] : memref<?x2xi32>
-// CHECK-NEXT:     affine.store %0, %arg2[0, 0] : memref<?x2xi32>
-// CHECK-NEXT:     %1 = affine.load %arg0[0, 1] : memref<?x2xi32>
-// CHECK-NEXT:     affine.store %1, %arg2[0, 1] : memref<?x2xi32>
-// CHECK-NEXT:     return
+// CHECK:    func.func @byval(%arg0: !llvm.struct<(i32, i32)>, %arg1: i32) -> !llvm.struct<(i32, i32)>
```

Comment:
Interesting, the original code put the return value as the last argument, and this change changes that behavior.

Reply:
`clang` returns `i64` instead of the pair. Does this violate the calling convention?

Reply:
We do not yet follow the ABI for struct returns. Compiling the test with -w, we can see a couple of warnings: 
```
cgeist -O0 --function='*' -S /localdisk2/etiotto/intel-llvm/polygeist/tools/cgeist/Test/Verification/pair.c
warning: struct should be flattened but MLIR codegen cannot yet handle it. Needs to be fixed.
warning: struct should be flattened but MLIR codegen cannot yet handle it. Needs to be fixed.
```
I think once we implement this part of the ABI correctly, we will get the LLVM IR clang generates (ABI conforming).

Reply:
Nice. Thanks, Ettore.